src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java

```diff
@@ -63,64 +76,139 @@ public abstract class AbstractDbt extends AbstractBash implements RunnableTask<S
     @PluginProperty(dynamic = true)
     String dbtPath = "./bin/dbt";
 
-    protected abstract java.util.List<String> commands(RunContext runContext) throws IllegalVariableEvaluationException;
+    @Schema(
+        title = "Input files are extra files that will be available in the dbt working directory.",
+        description = "You can define the files as map or a JSON string. " +
+            "Each file can be defined inlined or can reference a file from Kestra's internal storage."
+    )
+    @PluginProperty(
+        additionalProperties = String.class,
+        dynamic = true
+    )
+    private Object inputFiles;
 
-    @Override
-    public ScriptOutput run(RunContext runContext) throws Exception {
-        if (this.workingDirectory == null) {
-            this.workingDirectory = runContext.tempDir();
-        }
+    @Schema(
+        title = "The `profiles.yml` file content",
+        description = "If a `profile.yml` file already exist in the current working directory, setting this property will generate an error."
+    )
+    @PluginProperty(dynamic = true)
+    private String profiles;
 
-        ScriptOutput run = run(runContext, throwSupplier(() -> {
-            java.util.List<String> commands = new ArrayList<>(java.util.List.of(
-                runContext.render(dbtPath),
-                "--log-format json"
-            ));
+    // set RunnerType to PROCESS to keep backward compatibility as the old script engine has PROCESS by default and the new DOCKER
+    @Builder.Default
+    @Schema(
+        title = "Runner to use"
+    )
+    @PluginProperty
+    @NotNull
+    @NotEmpty
+    protected RunnerType runner = RunnerType.PROCESS;
 
-            if (this.projectDir != null) {
-                commands.add("--project-dir " + runContext.render(this.projectDir));
-            }
+    @Schema(
+        title = "Docker options for the `DOCKER` runner"
+    )
+    @PluginProperty
+    @Builder.Default
+    private DockerOptions docker = DockerOptions.builder()
+        .image("ghcr.io/dbt-labs/dbt-core")
```

Comment:
@anna-geller would prefer our docker image, (that would also remove the complexity of entrypoint)

Reply:
I thought we use the standard by default even if in example we use our own images;
As I understand it, for dbt, you should use an image with dependencies installed, so by default this should be an image with no dependencies as we don't know which one is needed that's why I uses dbt-core

Reply:
we can use DuckDB by default - it's super lightweight and doesn't require any external service: "ghcr.io/kestra-io/dbt-duckdb:latest"

Reply:
`hcr.io/kestra-io/dbt-duckdb:latest`

as in this blueprint:

```yaml
id: dbtGitDockerDuckDB
namespace: blueprint

tasks:
  - id: dbt
    type: io.kestra.core.tasks.flows.WorkingDirectory
    tasks:
    - id: cloneRepository
      type: io.kestra.plugin.git.Clone
      url: https://github.com/kestra-io/dbt-demo
      branch: main

    - id: dbt-build
      type: io.kestra.plugin.dbt.cli.DbtCLI
      runner: DOCKER
      docker:
        image: ghcr.io/kestra-io/dbt-duckdb:latest
      profiles: |
        jaffle_shop:
          outputs:
            dev:
              type: duckdb
              path: ":memory:"
              extensions: 
                - parquet
              fixed_retries: 1
              threads: 16
              timeout_seconds: 300
          target: dev
      commands:
        - dbt deps
        - dbt build
```
can even be added to unit tests

Reply:
dbt core is not useful for anything by itself without adapter and people often use the DuckDB dbt adapter for local/testing environments across all data warehouses (Snowflake, Reshift, BigQuery, etc) so it makes sense to use it as the default image

Reply:
I think we should not put any default as the user must select the image it needs

Reply:
fair, the example already uses one of our dbt images (`ghcr.io/kestra-io/dbt-bigquery:latest`) so can be left empty - LGTM! 👍

Reply:
good idea, let me check if Python will allow that :) could be dependency conflicts when trying to install all adapters in one package

Reply:
several dependency conflicts, you can see the logs https://github.com/kestra-io/examples/actions/runs/5871922953/job/15922419233 but the build succeeded and I'm able to import all the libraries:
![image](https://github.com/kestra-io/plugin-dbt/assets/86264395/827a3583-0d5e-45e9-af4c-d8bf1b7b11da)

so it should be OK to use `ghcr.io/kestra-io/dbt:latest` as the default image 👍

Reply:
Done